src/Avalonia.Base/Metadata/ItemTypeAttribute.cs

```diff
@@ -0,0 +1,19 @@
+﻿using System;
+namespace Avalonia.Metadata;
+
+/// <summary>
+/// Allows AvaloniaVS to identify the base element type of ItemsControl and derivatives
+/// </summary>
+[AttributeUsage(AttributeTargets.Class,
+    AllowMultiple = false,
+    Inherited = true)]
+public sealed class ItemTypeAttribute: DesignerAttribute
```

Comment:
In general, I am not sure about this PR.
Default controls could be guessed from the naming (ListBox -> ListBoxItem) or hardcoded. When this PR assumes that every custom control will apply this attribute to be efficient. Which is unlikely to be followed.

Reply:
`ItemTypeAttribute` is marked Inherited true. So if you create the `public class MyFashionListBox:ListBox`, it will inherit the ItemTypeAttribute from the `ListBox`.  Additionally we could create an analyzer/fixer or create a source generator that parses CreateContainerForItemOverride and generates ItemTypeAttribute.